pontoon/base/views.py

```diff
@@ -244,37 +287,33 @@ def translate(request, locale, slug, part=None, template='translate.html'):
 
     # Set subpage
     pages = Subpage.objects.filter(project=p)
-    setPart = False
     if len(pages) > 0:
         try:
             page = pages.get(name=part)
-            if pages.count() > 1:
-                setPart = True
 
         # If page not specified or doesn't exist
         except Subpage.DoesNotExist:
             page = pages[0]
-            if pages.count() > 1:
-                setPart = True
             locale_pages = pages.filter(resources__stats__locale=l)
             if locale_pages:
                 page = locale_pages[0]
-                setPart = True if locale_pages.count() > 1 else False
 
         data['page_url'] = page.url
-        if setPart:
-            data['part'] = page.name
+        data['part'] = page.name
 
-    # Set part if subpages not defined and entities in more than one file
+    # Set path if subpages not defined
     else:
         paths = (Resource.objects
                     .filter(project=p, entity_count__gt=0, stats__locale=l)
                     .order_by('path')
                     .values_list('path', flat=True))
 
-        if len(paths) > 1:
+        if len(paths) > 0:
             data['part'] = part if part in paths else paths[0]
 
+    # Set search keyword
+    data['search'] = request.GET.urlencode()
```

Comment:
I'm having a bit of trouble finding where you use this outside of checking if a search was submitted, which should be something you should be able to determine from the URL without having the server pass it back. Am I missing something?